shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -200,21 +225,81 @@ public FlutterEngine(
       @NonNull PlatformViewsController platformViewsController,
       @Nullable String[] dartVmArgs,
       boolean automaticallyRegisterPlugins) {
+    doInit(
+        context,
+        flutterLoader,
+        flutterJNI,
+        platformViewsController,
+        dartVmArgs,
+        automaticallyRegisterPlugins,
+        null);
+  }
+
+  public FlutterEngine() {}
```

Comment:
This is dangerous since it bifurcates the meaning of having a constructed engine. Rather, create an asynchronous factory that calls back with a FlutterEngine instance when it's ready. 

Generally, prefer composition over adding more functionalities and state to a single class.